warehouse/query-core/src/main/java/datawave/query/postprocessing/tf/FieldValue.java

```diff
@@ -0,0 +1,131 @@
+package datawave.query.postprocessing.tf;
+
+import datawave.util.StringUtils;
+import org.apache.accumulo.core.data.Key;
+import org.apache.hadoop.io.Text;
+
+/**
+ * A field name and value which is sorted on {@code <value>\0<name>}
+ */
+@Deprecated
+public class FieldValue implements Comparable<FieldValue> {
+    private int nullOffset;
+    private String valueField;
+    
+    public FieldValue(String field, String value) {
+        this.nullOffset = value.length();
+        this.valueField = value + '\0' + field;
+    }
+    
+    /**
+     * A distance between this field value and another. Here we want a distance that correlates with the number of keys between here and there for the same.
+     * Essentially we want the inverse of the number of bytes that match. document.
+     *
+     * @param fv
+     * @return a distance between here and there (negative means there is before here)
+     */
+    public double distance(FieldValue fv) {
+        byte[] s1 = getValueField().getBytes();
+        byte[] s2 = fv.getValueField().getBytes();
+        int len = Math.min(s1.length, s2.length);
+        
+        int matches = 0;
+        int lastCharDiff = 0;
+        
+        for (int i = 0; i <= len; i++) {
+            lastCharDiff = getValue(s2, i) - getValue(s1, i);
+            if (lastCharDiff == 0) {
+                matches++;
+            } else {
+                break;
+            }
+        }
+        
+        return Math.copySign(1.0d / (matches + 1), lastCharDiff);
+    }
+    
+    private int getValue(byte[] bytes, int index) {
+        if (index >= bytes.length) {
+            return 0;
+        } else {
+            return bytes[index];
+        }
+    }
+    
+    public String getValueField() {
+        return valueField;
+    }
+    
+    public String getField() {
```

Comment:
I'm curious, why not also store the value and field as separate attributes for this class? Then you won't need to perform the `substring()` call every time, and it will negate the need for `nullOffset`.